### PR TITLE
Time#to_json: include fraction part

### PR DESCRIPTION
**Pull request by perlun**
As agreed, we go with including the milliseconds only. If the time doesn't even have any fraction part, it will be ignored altogether.

### CODE REVIEW

**Comment by olleolleolle** [comment]
@perlun The build has lint errors.

**Comment by perlun** [comment]
> @perlun The build has lint errors.

It's not even just lint errors, the spec is broken. Working on a fix, thanks for the notice.

**Comment by perlun** [comment]
Should be fine for review now @olleolleolle

File: spec/ecraft/extensions/time_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      it 'serializes the date and time in ISO 8601 form, not including a fraction part' do
+        json = { 'time' => Time.new(2000, 1, 1, 0, 0, 0, '+02:00') }.to_json
+        expect(json).to eq '{"time":"2000-01-01T00:00:00+02:00"}'
+      end
+    end
+
+    context 'for values with a fraction part' do
+      it 'serializes the date and time in ISO 8601 form, including the fraction part' do
+        json = { 'time' => Time.new(2000, 1, 1, 1, 2, 3.399, '+02:00') }.to_json
+        expect(json).to eq '{"time":"2000-01-01T01:02:03.399+02:00"}'
```
Comment (olleolleolle) [inline spec/ecraft/extensions/time_spec.rb]: Spec style: we could rely more on context, to have room for some concrete details ("milliseconds").

```ruby
describe '#to_json' do
  context 'serializes date and time to ISO8601 form' do
    subject(:json) { { 'time' => time }.to_json }

    context 'with a fraction part' do
      let(:time) { Time.new(2000, 1, 1, 1, 2, 3.399, '+02:00') }

      it 'renders the milliseconds' do
        expect(json).to eq '{"time":"2000-01-01T01:02:03.399+02:00"}'
      end
    end

    context 'without a fraction part' do
      let(:time) { Time.new(2000, 1, 1, 0, 0, 0, '+02:00') }

      it 'renders the milliseconds' do
        expect(json).to eq '{"time":"2000-01-01T00:00:00+02:00"}'
      end
    end
  end
end
```

**Comment by perlun** [comment]
> Spec style: we could rely more on context, to have room for some concrete details ("milliseconds").

Thanks, adjusted in line with your changes. Will merge once Travis is done.

**Pull request merged**